tests/transforms/test_batch_inputs.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the batch inputs transform.
+"""
+import functools
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+def test_simple_circuit():
+    """Test that batching works for a simple circuit"""
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.batch_input(argnum=0)
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(inputs, weights):
+        qml.AngleEmbedding(inputs, wires=range(2), rotation="Y")
+        qml.RY(weights[0], wires=0)
+        qml.RY(weights[1], wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+    batch_size = 5
+    inputs = np.random.uniform(0, np.pi, (batch_size, 2))
+    inputs.requires_grad = False
+    weights = np.random.uniform(-np.pi, np.pi, (2,))
+
+    res = circuit(inputs, weights)
+    assert res.shape == (batch_size,)
+
+
+def test_value_error():
+    """Test if the batch_input raises relevant errors correctly"""
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.batch_input(argnum=0)
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(input1, input2, weights):
+        qml.AngleEmbedding(input1, wires=range(2), rotation="Y")
+        qml.RY(input2[0], wires=0)
+        qml.RY(weights[0], wires=0)
+        qml.RY(weights[1], wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+    batch_size = 5
+    input1 = np.random.uniform(0, np.pi, (batch_size, 2))
+    input1.requires_grad = False
+    input2 = np.random.uniform(0, np.pi, (4, 1))
+    input2.requires_grad = False
+    weights = np.random.uniform(-np.pi, np.pi, (2,))
+
+    try:
+        res = circuit(input1, input2, weights)
+    except ValueError as err:
+        pass
+    except Exception as err:
+        raise Exception(err)
```

Comment:
You can make use of `pytest.raises` here to check that the error message we expect is being raised :)

```suggestion
    with pytest.raises(ValueError, match="Batch dimension for all gate arguments"):
        res = circuit(input1, input2, weights)
```